packages/flutter_tools/lib/src/web/web_device.dart

```diff
@@ -128,7 +128,18 @@ abstract class ChromiumDevice extends Device {
   }) async {
     // See [ResidentWebRunner.run] in flutter_tools/lib/src/resident_web_runner.dart
     // for the web initialization and server logic.
-    final String url = platformArgs['uri']! as String;
+    String url;
+    if (debuggingOptions.webLaunchUrl != null) {
+      final RegExp pattern = RegExp(r'^((http)?:\/\/)[^\s]+');
+      if (pattern.hasMatch(debuggingOptions.webLaunchUrl!)) {
+        url = debuggingOptions.webLaunchUrl!;
+      } else {
+        url = platformArgs['uri']! as String;
+        _logger.printStatus('"${debuggingOptions.webLaunchUrl}" is not a vaild HTTP URL,"$url" will be used.');
```

Comment:
```suggestion
        _logger.printStatus('"${debuggingOptions.webLaunchUrl}" is not a valid HTTP URL,"$url" will be used.');
```

Reply:
should we tool exit in this case? the user might (will probably?) miss this print.

Reply:
Just don't want to interrupt the user's workflow.  If user miss this print, he can also spot the problem by the browser URL, as the picture shows,  and then configure a valid URL?
<img width="753" alt="Screen Shot 2021-12-10 at 3 41 17 PM" src="https://user-images.githubusercontent.com/4476512/145550764-cdcdb035-9156-435d-adf2-1ebaa6257e73.png">

If really think it's better to exit in this case,  I can modify it.

Reply:
> should we tool exit in this case? the user might (will probably?) miss this print.

@christopherfujino  Tool exit is as expected, I have fixed it, any other suggestions？